starboard/android/apk/app/src/app/AndroidManifest.xml

```diff
@@ -35,6 +35,9 @@
 
   <!-- https://iabtechlab.com/OTT-IFA, AdvertisingIdClient.Info.getId() -->
   <uses-permission android:name="com.google.android.gms.permission.AD_ID"/>
+
+  <!-- NPLB test needs to change the system timezone -->
+  <uses-permission android:name="android.permission.SET_TIME_ZONE"/>
```

Comment:
There is only 1 manifest for CoAT.

Ideally, we only need to change this for the manifest that is used by the app that is kicked by the test runner, this will block the usage of the method in the CoAT app though.

I 'd prefer to keep it as this. 
For production apps, Kimono does not need this. 
Apps cloned from the open source project should have their own manifest file, if they copy this file, they are on their own to remove it. We do not care about them too.

Reply:
This change will allows CoAT to call setTimeZone() in the future. Without the manifest change on google3, Kimono will fail though.
I think we should either 
1. add a comment on the top of the code to let people know when they wants to call this method, 
or 
2. change the manifest on google3 now.
Let me know what do you think.

Reply:
I am trying to find the log that run the test from the on_device test logs, but, I could not.

I ran this locally with test_runner, and found out it did not work. Here is the crash log. 

[ RUN      ] SbTimeZoneGetCurrentTest.IsKindOfSane
java.lang.SecurityException: setTimeZone: Neither user 10110 nor current process has android.permission.SET_TIME_ZONE.
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3011)
	at android.os.Parcel.createException(Parcel.java:2995)
	at android.os.Parcel.readException(Parcel.java:2978)
	at android.os.Parcel.readException(Parcel.java:2920)
	at android.app.IAlarmManager$Stub$Proxy.setTimeZone(IAlarmManager.java:343)
	at android.app.AlarmManager.setTimeZone(AlarmManager.java:1304)
	at dev.cobalt.coat.StarboardBridge.setTimeZone(StarboardBridge.java:855)
Caused by: android.os.RemoteException: Remote stack trace:
	at android.app.ContextImpl.enforce(ContextImpl.java:2240)
	at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:2268)
	at com.android.server.alarm.AlarmManagerService$5.setTimeZone(AlarmManagerService.java:2990)
	at android.app.IAlarmManager$Stub.onTransact(IAlarmManager.java:201)
	at android.os.Binder.execTransactInternal(Binder.java:1280)
--------- beginning of crash
Fatal signal 4 (SIGILL), code 1 (ILL_ILLOPC), fault addr 0xc25780f0 in tid 19016 (StarboardMain), pid 18940 (dev.cobalt.coat)
getWifiLinkLayerStats_1_6_Internal(l.1199) failed {.code = ERROR_UNKNOWN, .description = unknown error}
***Application Crashed***

I will look into how to get the permission for the test app.

Reply:
The test app has the permission set in the manifest file, see https://screenshot.googleplex.com/RaZm2joSSqE7pLj. However, the test still run into this android.permission.SET_TIME_ZONE permission error, so I think the API is not allowed to be used by the third party app as described on the doc https://screenshot.googleplex.com/Agn5FXVoQ8zYtgh.

Reply:
Late comment but i think it's maybe a conceptual mistake to try and route test framework features through Cobalt runtime. 

If a test can execute a simple ADB command to change the system timezone for a test, then why are we going through all this trouble in implementing this in C++ + Java code ?